### PR TITLE
Annotate System.Net.WebSockets.WebSocketProtocol for nullable reference types

### PR DESCRIPTION
**Pull request by stephentoub**
This also helped find that the annotation of `WebSocketReceiveResult(..., string? closeDescription)` was wrong in the ref (it was correct in the src), so I fixed that.
Contributes to https://github.com/dotnet/runtime/issues/2339
cc: @buyaa-n, @davidsh

### CODE REVIEW

File: src/libraries/System.Net.WebSockets.WebSocketProtocol/ref/System.Net.WebSockets.WebSocketProtocol.cs [hunk 1]
```diff
@@ -9,6 +9,6 @@ namespace System.Net.WebSockets
 {
     public static partial class WebSocketProtocol
     {
-        public static System.Net.WebSockets.WebSocket CreateFromStream(System.IO.Stream stream, bool isServer, string subProtocol, System.TimeSpan keepAliveInterval) { throw null; }
+        public static System.Net.WebSockets.WebSocket CreateFromStream(System.IO.Stream stream, bool isServer, string? subProtocol, System.TimeSpan keepAliveInterval) { throw null; }
```
Comment (buyaa-n) [inline src/libraries/System.Net.WebSockets.WebSocketProtocol/ref/System.Net.WebSockets.WebSocketProtocol.cs:12]: NIT: don't we want to add this annotation to the source to keep them same?

**Review by davidsh — APPROVED** [review]
LGTM. But there are still some nullable-related build errors.

**Review by buyaa-n — APPROVED** [review]
Left one NIT, overall looks good thanks! 

JFYI: build failures on src\libraries\Common\src\System\Net\WebSockets\ManagedWebSocket.cs because Debug.Assert(...) not suppressing nullable warnings now

File: src/libraries/System.Net.WebSockets.WebSocketProtocol/ref/System.Net.WebSockets.WebSocketProtocol.cs (hunk 1 above)
Comment (stephentoub) [reply src/libraries/System.Net.WebSockets.WebSocketProtocol/ref/System.Net.WebSockets.WebSocketProtocol.cs:12]: Oops, yes, I neglected to commit that file.

**Pull request merged**